src/main/scala/org/deepdive/ddlog/DeepDiveLogParser.scala

```diff
@@ -253,8 +248,15 @@ class DeepDiveLogParser extends JavaTokenParsers {
       }
     )
 
-  def functionMode = "mode" ~> "=" ~> functionModeType
-  def inferenceMode = "mode" ~> "=" ~> inferenceModeType
+  def ruleAnnotation = "@" ~> annotationName ~ ("(" ~> rep1sep(ident, ",") <~ ")") ^^ {
+    case (name ~ args) => RuleAnnotation(name, args)
+  }
+
+  def ruleAnnotations = rep(ruleAnnotation)
```

Comment:
In 68581a3df432c4c9be5c263dc1dc697a1d834418 I tried the general annotation for extraction and inference rules, but it seems awkward, so in cf5759fab85cfd507d6abaf6b9c25ad71a0e56cd I switched back to the current style. These lines are for future use of rule annotations.